File: Remove Element/RemoveElement.py
class RemoveElement(object):
    def removeElement(self, nums, val):
        if nums is None or len(nums) == 0:
            return 0
        currentIndex = -1
        nextIndex = 0
        while nextIndex < len(nums):
            if nums[nextIndex] != val:
                currentIndex +=1
                nums[currentIndex] = nums[nextIndex]
            nextIndex+=1
        return currentIndex + 1

File: 0027_Remove Element/RemoveElement.cs
public class Solution {
    public int RemoveElement(int[] nums, int val) {
        if(nums == null || nums.Length ==0) return 0;
        var s = 0;
        var e = 0;
        while(e < nums.Length){
            if(nums[e] != val){
                nums[s++] = nums[e];
            }
            
            e++;
        }
        
        return s;
    }
}